### ISSUE TITLE
feature: support the (X/Y) display mode for the printcolumn field in CR resource

### ISSUE THREAD

**Issue by googs1025** [kubernetes/kubernetes#127246]
### What would you like to be added?

When I was working on implementing a CR controller using kubebuilder, I wanted to implement a way to display columns using a fraction format (i.e. X/Y).
```go
// +genclient
...
//+kubebuilder:printcolumn:name="Ready",type="string",JSONPath="{.status.readyReplicas}/{.spec.replicas}",description="Ratio of ready replicas to desired replicas."
//+kubebuilder:printcolumn:name="UP-TO-DATE",type="string",JSONPath=".status.updatedReplicas",description="Number of groups that have been updated (ready or not)."
//+kubebuilder:printcolumn:name="Age",JSONPath=".metadata.creationTimestamp",type=date,description="Age is the time LeaderWorkerSet was created."

// LeaderWorkerSet is the Schema for the leaderworkersets API
type LeaderWorkerSet struct {
	...
}
```

`//+kubebuilder:printcolumn:name="Ready",type="string",JSONPath="{.status.readyReplicas}/{.spec.replicas}",description="Ratio of ready replicas to desired replicas."
` 

I used the following method and found that an error would be reported. The error is as follows:
```bash
root@VM-0-8-ubuntu:/home/ubuntu/lws# make uninstall
make: go: Permission denied
test -s /home/ubuntu/lws/bin/controller-gen && /home/ubuntu/lws/bin/controller-gen --version | grep -q v0.16.2 || \
GOBIN=/home/ubuntu/lws/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.16.2
/home/ubuntu/lws/bin/controller-gen \
	rbac:roleName=manager-role output:rbac:artifacts:config=config/rbac \
	crd:generateEmbeddedObjectMeta=true output:crd:artifacts:config=config/crd/bases \
	webhook output:webhook:artifacts:config=config/webhook \
	paths="./..."
/home/ubuntu/lws/bin/kustomize build config/crd | kubectl delete --ignore-not-found=false -f -
customresourcedefinition.apiextensions.k8s.io "leaderworkersets.leaderworkerset.x-k8s.io" deleted
root@VM-0-8-ubuntu:/home/ubuntu/lws# make install
make: go: Permission denied
test -s /home/ubuntu/lws/bin/controller-gen && /home/ubuntu/lws/bin/controller-gen --version | grep -q v0.16.2 || \
GOBIN=/home/ubuntu/lws/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.16.2
/home/ubuntu/lws/bin/controller-gen \
	rbac:roleName=manager-role output:rbac:artifacts:config=config/rbac \
	crd:generateEmbeddedObjectMeta=true output:crd:artifacts:config=config/crd/bases \
	webhook output:webhook:artifacts:config=config/webhook \
	paths="./..."
/home/ubuntu/lws/bin/kustomize build config/crd | kubectl create -f -
The CustomResourceDefinition "leaderworkersets.leaderworkerset.x-k8s.io" is invalid: spec.additionalPrinterColumns[0].JSONPath: Invalid value: "{.status.readyReplicas}/{.spec.replicas}": must be a simple json path starting with .
make: *** [Makefile:187: install] Error 1
```
When I asked for help from the kubebuilder project, it seemed to ask me to submit an issue to controller-tool https://github.com/kubernetes-sigs/kubebuilder/discussions/4134#discussioncomment-10575562 , such as:
When I submitted an issue to controller-tool, it seemed to say that this was not a function that this project could provide. https://github.com/kubernetes-sigs/controller-tools/issues/1051#issuecomment-2337525823

I don't understand how it's generated. Therefore, I will submit an issue here for discussion (if it is not a function that can be provided by kubernetes/kubernetes, I will close this issue)

I don't know if my usage is wrong or if JsonPath itself only provides simple field parsing.

In addition,is there a way we could support a MultiJsonPath-like approach, allowing multiple JsonPaths to be combined? This would make the display more flexible, rather than being limited to specific JsonPaths only.

My expectation is 3/3. like this
```go
root@VM-0-8-ubuntu:/home/ubuntu/lws/config/samples# kubectl get lws
NAME                     READY   UP-TO-DATE   AGE
leaderworkerset-sample    3/3      3          6s
```

### Why is this needed?

- Make the display of CR resource objects in kubectl commands more flexible

**Comment by googs1025** [comment]
/kind support
Sorry, I don't know which sig or wg this question belongs to

**Comment by damsien** [comment]
Currently in a CRD, we only use JSON path **expression**. To solve this issue, we need to to use JSON path **template** instead. I propose the following: either we change the `JSONPath` field behavior in a new version of the CRD object, or we add a `JSONPathTemplate` field in the `CustomResourceColumnDefinition` type.

After discussion about the implementation, I would like to take this is issue if possible.

**Comment by cici37** [comment]
Hi @damsien , thanks for the volunteer. The proposal you raised involves API change and would be great to be discussed before raising the PR. Would you be interested in taking up this one? If so, the discussion would be brought to sig slack channel or sig meeting. Thank you

**Comment by damsien** [comment]
Yes I would like to take this one. I'll start the discussion on the `sig-api-machinery` slack channel.

**Comment by sreeram-venkitesh** [comment]
Related: @Priyankasaggu11929 and I are working on adding CEL support for additionalPrinterColumns ([KEP](https://github.com/kubernetes/enhancements/issues/4595)). This is another possible solution for this requirement.

**Comment by cici37** [comment]
Hi @googs1025 , Thanks for raising the issue. We talked about the similar topic within sig apimachinery and the general agreement would be that the value added with only supporting certain format(e.g. X/Y) is not as much as supporting a general mechanism. The relevant topic under discussion currently is what mentioned above (CEL support for additionalPrinterColumns). Maybe talk to @sreeram-venkitesh and @Priyankasaggu11929 to see if the use case could be covered :) Thank you

**Comment by damsien** [comment]
Hello @sreeram-venkitesh @Priyankasaggu11929 ,

Hope you're doing great :)
The KEP has been open for nearly 2 years and I was wondering if you are still working on it? Also, I would be happy to help contributing on this one if any help is needed (or if don't have the time) 😊
